### PR TITLE
Bo b 14 create randomized map

### PR DESCRIPTION
**Pull request by spirit126**
Add RNG Map Generator

- bit map generator
    - single room protector
    - smoother
    - corner smoother
- player spawner
- spawn Point spawner

### CODE REVIEW

File: Assets/Scripts/Attributes/HealthSystemAttribute.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ private void Start()
         // Find the UI in the scene and store a reference for later use
         ui = GameObject.FindObjectOfType<UIScript>();
 
-        if (CompareTag("Player"))
+        if (CompareTag(tag: "Player"))
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/Attributes/HealthSystemAttribute.cs:19]: nessecary?

Comment (SomethingWittyWithATechAspect) [reply Assets/Scripts/Attributes/HealthSystemAttribute.cs:19]: only commented once

File: Assets/Scripts/CrosshairController.cs [hunk 2]
```diff
@@ -6,8 +6,11 @@ public class CrosshairController : MonoBehaviour
 
     void Start()
     {
-            //Hide cursor for Production Builds 
-            Cursor.visible = Debug.isDebugBuild;
+            #if DEBUG
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/CrosshairController.cs:9]: does unity has a way to determine when something is production build?

File: Assets/Scripts/MapTools/Door.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using UnityEngine;
+
+namespace MapTools
+{
+    public class Door : MonoBehaviour
+    {
+        private void OnTriggerEnter2D(Collider2D col)
+        {
+            if (col.CompareTag("Player"))
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/MapTools/Door.cs:9]: why no tag: Player here?

File: Assets/Scripts/MapTools/MapMatrixGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                        }
+                    }
+                    else
+                    {
+                        wallCount++;
+                    }
+                }
+            }
+
+            Console.WriteLine(wallCount);
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/MapTools/MapMatrixGenerator.cs:101]: remove Console.write

File: Assets/Scripts/SpawnPoint.cs [hunk 5]
```diff
@@ -29,6 +36,8 @@ void Update()
     private void SpawnEnemy()
     {
         Vector3 position = transform.position;
-        Instantiate(enemies[_random.Next(enemies.Length - 1)], position, Quaternion.Euler(new Vector3(0f, 0f, 0f)));
+        var map = GameObject.Find("Map");
+        var mapController = (MapGenerator) map.GetComponent(typeof(MapGenerator));
+        mapController.AddEnemy((int)position.x, (int)position.y);
     }
 }
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/SpawnPoint.cs:43]: General question regarding coding guidelines 
Do we do new line at end of file?
@zicons

**Review by SomethingWittyWithATechAspect — CHANGES_REQUESTED** [review]
Also Tests missing for Map Gen, Seed Gen and Door

File: Assets/Scripts/Attributes/HealthSystemAttribute.cs (hunk 1 above)
Comment (spirit126) [reply Assets/Scripts/Attributes/HealthSystemAttribute.cs]: no and fixed

File: Assets/Scripts/CrosshairController.cs (hunk 2 above)
Comment (spirit126) [reply Assets/Scripts/CrosshairController.cs]: no unity do not have this option in the preprocessor. Fixed to unity method

File: Assets/Scripts/MapTools/MapMatrixGenerator.cs (hunk 4 above)
Comment (spirit126) [reply Assets/Scripts/MapTools/MapMatrixGenerator.cs]: removed

File: Assets/Scripts/SpawnPoint.cs (hunk 5 above)
Comment (spirit126) [reply Assets/Scripts/SpawnPoint.cs:43]: yes, and fixed

File: Assets/AltUnityTester/Editor/Tests/MapTools/DoorTest.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            altUnityDriver.Stop();
+        }
+
+        [SetUp]
+        public void Reset()
+        {
+            first = true;
+        }
+    }
+}
```
Comment (SomethingWittyWithATechAspect) [inline Assets/AltUnityTester/Editor/Tests/MapTools/DoorTest.cs:37]: end of file new line

File: Assets/Scripts/SpawnPoint.cs (hunk 5 above)
Comment (SomethingWittyWithATechAspect) [reply Assets/Scripts/SpawnPoint.cs:43]: you forgot new line here

**Review by SomethingWittyWithATechAspect — CHANGES_REQUESTED** [review]
2 end of file new lines missing

**Review by SomethingWittyWithATechAspect — APPROVED** [review]

**Pull request merged**